src/Compilers/Core/CodeAnalysisTest/AsyncQueueTests.cs

```diff
@@ -170,41 +170,31 @@ public async Task DequeueAsyncWithCancellation()
         }
 
         [Fact]
-        public async Task DequeueAsyncWithCancellationAfterComplete()
+        public async Task EnqueueAfterDequeueAsyncWithCancellation()
         {
             var queue = new AsyncQueue<int>();
             var cts = new CancellationTokenSource();
             var task = queue.DequeueAsync(cts.Token);
-            Assert.False(task.IsCompleted);
-            queue.Enqueue(42);
-            await task.ConfigureAwait(false);
+            Assert.False(task.IsCanceled);
             cts.Cancel();
+            await Assert.ThrowsAsync<TaskCanceledException>(() => task);
+            Assert.Equal(TaskStatus.Canceled, task.Status);
+
+            queue.Enqueue(1);
```

Comment:
📝 Prior to this change, this `Enqueue` value would get assigned to a waiter which would never be observed.